src/ActionsObservable.js

```diff
@@ -23,6 +25,14 @@ export class ActionsObservable extends Observable {
     return observable;
   }
 
+  catchAndRelease(action) {
+    return this::_catch((error, source) => {
+      const fullAction = Object.assign({}, action, { error });
```

Comment:
How bout it's just agnostic of actions in general:

```js
onErrorReturnAndRetry(selector) {
  return this::_catch((error, source) => {
    const value = selector(error);
    return merge(of(value), source);
  });
}
```

```js
const somethingErrored = error => ({ type: SOMETHING_ERRORED, error });

action$.ofType(SOMETHING)
  .switchMap(...)
  .onErrorReturnAndRetry(error => somethingErrored(error))
```

Or does that make it too less useful compared to just doing the catch/of/merge yourself?